src/site/stages/build/plugins/check-broken-links/helpers/isBrokenLink.js

```diff
@@ -0,0 +1,34 @@
+const path = require('path');
+const url = require('url');
+
+function isBrokenLink(link, pagePath, allPaths) {
+  if (!link) return true;
+
+  const parsed = url.parse(link);
+  const isExternal = !!parsed.protocol;
+
+  if (isExternal) return false;
+
+  if (!parsed.pathname) {
+    // Not empty, not external, but also no path.
+    // This has to be an hash link or query param change.
+
+    return false;
+  }
+
+  let filePath = decodeURI(parsed.pathname);
+
+  if (path.isAbsolute(filePath)) {
+    filePath = path.join('.', filePath);
+  } else {
+    filePath = path.join(pagePath, filePath);
+  }
+
+  if (!path.extname(filePath)) {
+    filePath = path.join(filePath, 'index.html');
+  }
+
+  return !allPaths.has(filePath);
+}
+
+module.exports = isBrokenLink;
```

Comment:
`getBrokenLinks.js` parses the links out of every HTML page, processing each link value (an HREF or a SRC) through this function to determine whether it points to a valid file or not.